Source/Core/Core/Scripting/HelperClasses/GCButton_API_Implementations.cpp

```diff
@@ -0,0 +1,292 @@
+#include "Core/Scripting/HelperClasses/GCButton_API_Implementations.h"
+
+#include "Core/Scripting/CoreScriptInterface/Enums/GCButtonNameEnum.h"
+
+#include "string.h"
+
+namespace Scripting
+{
+
+static bool IsEqualIgnoreCase(const char* string_1, const char* string_2)
+{
+  int index = 0;
+  while (string_1[index] != '\0')
+  {
+    if (string_1[index] != string_2[index])
+    {
+      char first_char = string_1[index];
+      if (first_char >= 65 && first_char <= 90)
+        first_char += 32;
+      char second_char = string_2[index];
+      if (second_char >= 65 && second_char <= 90)
+        second_char += 32;
+      if (first_char != second_char)
+        return false;
+    }
+    ++index;
+  }
+
+  return string_2[index] == '\0';
+}
```

Comment:
Feels odd that you _have to_ implement this by hand. What about using `strcasecmp` and its friends?

Depending on _where else_ you need this, we got `StringUtil::CaseInsensitiveEquals`, but it takes a `string_view` as parameter (and not a `char*`)

Reply:
@BhaaLseN This is the one change you suggested that I couldn't resolve, since strcasecmp isn't a standard C++ function (and isn't available from Dolphin by default without adding in other imports).

Reply:
`const char*` should be able to be passed as an `std::string_view`.

Reply:
Added in those change, so this issue is now resolved.